packages/cli/generators/controller/index.js

```diff
@@ -34,25 +65,150 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
+      });
+  }
+
+  promptArtifactCrudVars() {
+    let modelList = [];
+    let repositoryList = [];
+    if (
+      !this.artifactInfo.controllerType ||
+      this.artifactInfo.controllerType === ControllerGenerator.BASIC
+    ) {
+      return;
+    }
+    return utils
+      .getArtifactList(this.artifactInfo.modelDir, 'model')
+      .then(list => {
+        if (_.isEmpty(list)) {
+          return this.exit(
+            new Error(`No models found in ${this.artifactInfo.modelDir}`)
+          );
+        }
+        modelList = list;
+        return utils.getArtifactList(
+          this.artifactInfo.repositoryDir,
+          'repository',
+          true
+        );
+      })
+      .then(list => {
+        if (_.isEmpty(list)) {
+          return this.exit(
+            new Error(
+              `No repositories found in ${this.artifactInfo.repositoryDir}`
+            )
+          );
+        }
+        repositoryList = list;
+        return this.prompt([
+          {
+            type: 'list',
+            name: 'modelName',
+            message:
+              'What is the name of the model to use with this CRUD repository?',
+            choices: modelList,
+            when: this.artifactInfo.modelName === undefined,
+            default: modelList[0],
+            validate: utils.validateClassName,
+          },
+          {
+            type: 'list',
+            name: 'repositoryName',
+            message: 'What is the name of your CRUD repository?',
+            choices: repositoryList,
+            when: this.artifactInfo.repositoryName === undefined,
+            default: repositoryList[0],
+            validate: utils.validateClassName,
+          },
+          {
+            type: 'list',
+            name: 'id',
+            message: 'What is the type of your ID?',
+            choices: ['number', 'string', 'object'],
+            when: this.artifactInfo.id === undefined,
+            default: 'number',
+          },
+        ]).then(props => {
+          debug(`props: ${inspect(props)}`);
+          Object.assign(this.artifactInfo, props);
+          // Ensure that the artifact names are valid.
+          [
+            this.artifactInfo.name,
+            this.artifactInfo.modelName,
+            this.artifactInfo.repositoryName,
+          ].forEach(item => {
+            item = utils.toClassName(item);
+          });
+          // Create camel-case names for variables.
+          this.artifactInfo.repositoryNameCamel = utils.camelCase(
+            this.artifactInfo.repositoryName
+          );
+          this.artifactInfo.modelNameCamel = utils.camelCase(
+            this.artifactInfo.modelName
+          );
+          return props;
+        });
+      });
+  }
+
   scaffold() {
-    super.scaffold();
+    // We don't want to call the base scaffold function since it copies
+    // all of the templates!
     if (this.shouldExit()) return false;
+    this.artifactInfo.name = utils.toClassName(this.artifactInfo.name);
```

Comment:
Is this necessary given we did it in lines 155-160 above? (https://github.com/strongloop/loopback-next/pull/849/files#diff-3195cd2d1a3887230542f8d33260e203R160)

Reply:
Yes, since that step is not guaranteed to run (basic controllers).